pandas/tests/test_lib.py

```diff
@@ -227,6 +227,16 @@ def test_maybe_convert_numeric_infinities(self):
                         np.array(['foo_' + infinity], dtype=object),
                         na_values, maybe_int)
 
+    def test_maybe_convert_numeric_post_floatify_nan(self):
+        # see gh-13314
+        data = np.array(['1.200', '-999.000', '4.500'], dtype=object)
```

Comment:
can you move the like tests to a separate class (TestInference), rather than TestMisc